orchagent/copporch.cpp

```diff
@@ -0,0 +1,540 @@
+        if (!getTrapID(trap_id_str, trap_id))
+        {
+            SWSS_LOG_ERROR("Invalid trap_id value specified %s\n", trap_id_str.c_str());
+            return false;
+        }
+        SWSS_LOG_DEBUG("Input trap_id:%s, pushing trap_id:%d", trap_id_str.c_str(), trap_id);
+        trap_id_list.push_back(trap_id);
+    }
+    return true;
+}
+
+bool CoppOrch::applyTrapIds(sai_object_id_t trap_group, vector<string> &trap_id_name_list, std::vector<sai_attribute_t> &trap_id_attribs)
+{
+    SWSS_LOG_ENTER();
+    sai_status_t sai_status;
+    vector<sai_hostif_trap_id_t> trap_id_list;
+    if (!getTrapIdList(trap_id_name_list, trap_id_list))
+    {
+        return false;
+    }
+    sai_attribute_t attr;
+    attr.id = SAI_HOSTIF_TRAP_ATTR_TRAP_GROUP;
+    attr.value.oid = trap_group;
+    for (auto trap_id : trap_id_list)
+    {
+        SWSS_LOG_DEBUG("set trap attr.1:%d", attr.id);
+        if (SAI_STATUS_SUCCESS != (sai_status = sai_hostif_api->set_trap_attribute(trap_id, &attr)))
+        {
+            SWSS_LOG_ERROR("Failed to apply trap_id:%d to trap group:%llx, error:%d\n", trap_id, trap_group, sai_status);
+            return false;
+        }
+        for(auto trap_id_attr : trap_id_attribs)
+        {
+            SWSS_LOG_DEBUG("set trap attr.2:%d", trap_id_attr.id);
+            if (SAI_STATUS_SUCCESS != (sai_status = sai_hostif_api->set_trap_attribute(trap_id, &trap_id_attr)))
+            {
+                SWSS_LOG_ERROR("Failed to apply trap_id attribute:%d to trap_id:%d, error:%d\n", trap_id_attr.id, trap_id, sai_status);
+                return false;
+            }
+        }
+    }
+    return true;    
+}
+
+bool CoppOrch::removePolicerFromTrapGroup(sai_object_id_t trap_group)
+{
+    sai_attribute_t attr;
+    sai_status_t sai_status;
+    sai_object_id_t policer_id;    
+    attr.id = SAI_HOSTIF_TRAP_GROUP_ATTR_POLICER;
+    attr.value.oid = SAI_NULL_OBJECT_ID;
+    if (SAI_STATUS_SUCCESS != (sai_status = sai_hostif_api->get_trap_group_attribute(trap_group, 1, &attr))) {
+        SWSS_LOG_ERROR("Failed to get policer for trap gorup:%llx, error:%d\n", trap_group, sai_status);
+        return false;
+    }
+    policer_id = attr.value.oid;
+    if(SAI_NULL_OBJECT_ID == policer_id)
+    {
+        SWSS_LOG_DEBUG("No policer is attached to trap gorup:%llx\n", trap_group);
+        return true;
+    }
+
+    attr.value.oid = SAI_NULL_OBJECT_ID;
+    if (SAI_STATUS_SUCCESS != (sai_status = sai_hostif_api->set_trap_group_attribute(trap_group, &attr))) {
+        SWSS_LOG_ERROR("Failed to reset policer for trap gorup:%llx, error:%d\n", trap_group, sai_status);
+        return false;
+    }
+    if (SAI_STATUS_SUCCESS != (sai_status = sai_policer_api->remove_policer(policer_id)))
+    {
+        SWSS_LOG_ERROR("Failed to remove policer:%llx, error:%d\n", policer_id, sai_status);
+        return false;
+    }
+    return true;
+}
+task_process_status CoppOrch::processCoppRule(Consumer& consumer)
```

Comment:
keep a blank line between functions. there are a few other places to show similiar issue, can you address all of them?